packages/message/src/StepFunctions.ts

```diff
@@ -150,15 +150,15 @@ export const getFailedStepName = (
 /**
  * Finds all failed execution events and returns the last one in the list.
  *
- * @param {Array<HistoryEventList>} events - array of AWS Stepfunction execution HistoryEvents
- * @returns {HistoryEventList | undefined} - the last lambda or activity that failed in the
+ * @param {HistoryEvent[]} events - array of AWS Stepfunction execution HistoryEvents
+ * @returns {HistoryEvent[] | undefined} - the last lambda or activity that failed in the
  * event array, or an empty array.
  */
 export const lastFailedEventStep = (
-  events: AWS.StepFunctions.HistoryEvent[]
-): AWS.StepFunctions.HistoryEvent | undefined => {
+  events: HistoryEvent[]
+): HistoryEvent | undefined => {
   const failures = events.filter((event) =>
-    ['LambdaFunctionFailed', 'ActivityFailed'].includes(event.type));
+    ['LambdaFunctionFailed', 'ActivityFailed'].includes(event.type as string));
```

Comment:
Why the cast?

Reply:
event.type is a HistoryEvent. In order to compare it to a list of strings, the cast is necessary. I think this was also from Charles, but that is my understanding of why.